engine/src/test/java/com/redhat/cloud/notifications/MicrometerAssertionHelper.java

```diff
@@ -0,0 +1,85 @@
+package com.redhat.cloud.notifications;
+
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Timer;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import java.time.Duration;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+import static org.awaitility.Awaitility.await;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+/**
+ * {@link MeterRegistry#clear()} cannot be called between tests to reset counters because it would remove the counters
+ * instances from the registry while we use many {@link ApplicationScoped} beans which hold references pointing at the
+ * aforementioned counters instances. That's why we need this helper to assert increments in a reliable way.
+ */
+@ApplicationScoped
+public class MicrometerAssertionHelper {
```

Comment:
Copied from `backend/.../MicrometerAssertionHelper`, code unchanged. The `backend` class will be completely removed eventually, but not in this PR.